### ISSUE TITLE
Rule: No concatenation with __dirname, __filename

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#540]
This rule would be Node.js-specific and would flag patterns such as:

``` js
var myPath = __dirname + "/app/foo.js";
```

Concatenating strings with `__dirname` or `__filename` isn't a good idea due to OS-specific file paths and possibly improperly resolving the filename. It's better to use the`path.join()` method.

The rule ID should be `no-dir-concat` and should be on by default for the `node` environment only.
